custom_components/powercalc/config_flow.py

```diff
@@ -0,0 +1,580 @@
+            self.name = user_input.get(CONF_NAME)
+            self.sensor_config.update(user_input)
+            if not errors:
+                return self.create_config_entry()
+
+        return self.async_show_form(
+            step_id="group",
+            data_schema=_create_group_schema(self.hass, SCHEMA_GROUP),
+            errors=errors,
+        )
+    
+    async def async_step_fixed(self, user_input: dict[str,str] = None) -> FlowResult:
+        errors = {}
+        if user_input is not None:
+            self.sensor_config.update({CONF_FIXED: user_input})
+            errors = await self.validate_strategy_config()
+            if not errors:
+                return self.create_config_entry()
+
+        return self.async_show_form(
+            step_id="fixed",
+            data_schema=SCHEMA_POWER_FIXED,
+            errors=errors,
+        )
+    
+    async def async_step_linear(self, user_input: dict[str,str] = None) -> FlowResult:
+        errors = {}
+        if user_input is not None:
+            self.sensor_config.update({CONF_LINEAR: user_input})
+            errors = await self.validate_strategy_config()
+            if not errors:
+                return self.create_config_entry()
+
+        return self.async_show_form(
+            step_id="linear",
+            data_schema=_create_linear_schema(self.source_entity_id),
+            errors=errors,
+        )
+    
+    async def async_step_wled(self, user_input: dict[str,str] = None) -> FlowResult:
+        errors = {}
+        if user_input is not None:
+            self.sensor_config.update({CONF_WLED: user_input})
+            errors = await self.validate_strategy_config()
+            if not errors:
+                return self.create_config_entry()
+
+        return self.async_show_form(
+            step_id="wled",
+            data_schema=SCHEMA_POWER_WLED,
+            errors=errors,
+        )
+
+    async def async_step_lut(self, user_input: dict[str,str] = None) -> FlowResult:
+        """Try to autodiscover manufacturer/model first. Ask the user to confirm this or forward to manual configuration"""
+        if user_input is not None:
+            if user_input.get(CONF_CONFIRM_AUTODISCOVERED_MODEL):
+                return self.create_config_entry()
+
+            return await self.async_step_lut_manufacturer()
+
+        model_info = await autodiscover_model(self.hass, self.source_entity.entity_entry)
+        if model_info:
+            return self.async_show_form(
+                step_id="lut",
+                description_placeholders={
+                    "manufacturer": model_info.manufacturer,
+                    "model": model_info.model,
+                },
+                data_schema=SCHEMA_POWER_LUT_AUTODISCOVERED,
+                errors={},
+            )
+
+        return await self.async_step_lut_manufacturer()
+
+    async def async_step_lut_manufacturer(self, user_input: dict[str,str] = None) -> FlowResult:
```

Comment:
Do you need to handle the case of old version didn't detect model -> user manually set up -> now available automatically; prompt or warn?
(This method isn't about doing that, but it occurred to me when reading)

Reply:
More I guess, will manually configured items be always excluded from future configflow prompts/auto detection? I wasn't sure from reading the code alone.

Reply:
The lights you already setup will not be excluded from the list when you setup a new one. The user is free to setup multiple power sensors for the same light/entity as long as he gives it a unique_id. When an entry already exists with the same uniqueID the config flow will abort with an error.

The entries which are setup by the user with either YAML of with the GUI config flow always have preference over the auto detection algorithm. When the auto detection routine discovered that `light.bathroom` can be setup without configuration and the user creates a config entry for this same light, than the auto detected one is just skipped.

> I add a light, someone else later measures it, I upgrade, should I get warnings or advice?

That's a good one, did not thought about this. It's out of scope for this first version. This same issue applies to manually setup sensors with the YAML.
Would be a nice feature to add in the future, but not a real easy one to implement.